app/controllers/application_controller.rb

```diff
@@ -2,4 +2,13 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  protected
+  def authenticate_user!
```

Comment:
For style consistency, next this method one level further underneath `protected`. Also, make sure to add a flash message indicating to the user why they were redirected.